pkg/operator/controller/garden/components.go

```diff
@@ -671,3 +676,21 @@ func (r *Reconciler) newGardenerAccess(secretsManager secretsmanager.Interface,
 		},
 	)
 }
+
+func (r *Reconciler) newNginxIngressController() (component.DeployWaiter, error) {
+	return sharedcomponent.NewNginxIngress(
+		r.RuntimeClientSet.Client(),
+		r.ImageVector,
+		r.RuntimeVersion,
+		v1beta1constants.NginxIngressClass,
+		map[string]string{
+			"enable-vts-status":            "false",
+			"server-name-hash-bucket-size": "256",
+			"use-proxy-protocol":           "false",
+			"worker-processes":             "2",
+			"allow-snippet-annotations":    "false",
+		},
+		nil,
+		r.GardenNamespace,
+		"system-cluster-critical")
```

Comment:
Use the constant for `gardener-garden-system-300` and maintain it in https://github.com/gardener/gardener/blob/master/docs/development/priority-classes.md#priorityclasses-for-garden-control-plane-components.